src/cbmc/cbmc_parse_options.cpp

```diff
@@ -904,6 +905,13 @@ bool cbmc_parse_optionst::process_goto_program(
     status() << "Partial Inlining" << eom;
     goto_partial_inline(goto_functions, ns, ui_message_handler);
 
+    if(cmdline.isset("string-refine"))
+    {
+      status() << "Preprocessing for java Character library" << eom;
```

Comment:
@kroening @peterschrammel @romainbrenguier I am wondering whether the invoked code should go in java_bytecode. (Yes, I have asked a similar question before.) If that folder isn't a good fit (for it introduces a dependency on goto-programs from java_bytecode), then maybe it's time for some folder like java_in-processing?